apiwasm/file/fileguest.go

```diff
@@ -43,3 +43,13 @@ func (f *myFileSvc) LoadTestData(ctx context.Context, in *file.LoadTestDataReque
 	return file.LoadTestDataHost(in)
 
 }
+
+func (f *myFileSvc) Read(ctx context.Context, in *file.ReadRequest) (*file.ReadResponse, file.FileErr) {
+	return file.ReadHost(in)
```

Comment:
This idea is something you should decide on, it's your call:

Is it better to return your own result type (`ReadResponse`) or just return what `Read()` on a `Reader` already does.  I think the Read() is defined to return the number of bytes read--and its put them into a buffer the caller supplies the buffer to write into.

You can make the argument that using your own return type gives you more control... but using the same setup as a Read() has the advantage that any go programmer will already know how to use it.

Your call.

Reply:
yeah, Read() takes in buf and return the number of bytes read and error.

so I wrapped buf in ReadRequest and number of bytes read in ReadResponse. I feel like this kind of trade off makes both sides feel more comfortable. Do you have any insights?